device/sycl/src/clusterization/clusterization_algorithm.cpp

```diff
@@ -18,32 +18,31 @@
 #include "spacepoint_formation.hpp"
 
 // Vecmem include(s).
-#include <vecmem/utils/copy.hpp>
+#include <vecmem/utils/sycl/copy.hpp>
 
 // System include(s).
 #include <algorithm>
 
 namespace traccc::sycl {
 
-clusterization_algorithm::clusterization_algorithm(vecmem::memory_resource &mr,
+clusterization_algorithm::clusterization_algorithm(traccc::memory_resource& mr,
                                                    queue_wrapper queue)
     : m_mr(mr), m_queue(queue) {}
 
 clusterization_algorithm::output_type clusterization_algorithm::operator()(
-    const cell_container_types::host &cells_per_event) const {
+    const cell_container_types::host& cells_per_event) const {
 
     // Number of modules
     unsigned int num_modules = cells_per_event.size();
 
     // Vecmem copy object for moving the data between host and device
-    vecmem::copy copy;
+    vecmem::sycl::copy copy(m_queue.queue());
```

Comment:
:thinking: When using `vecmem::sycl::shared_memory_resource` I believe we need to stick to `vecmem::copy`.

You should add an `std::unique_ptr<vecmem::copy>` member to the algorithm, and initialise it with either `vecmem::copy` or `vecmem::sycl::copy` based on the values in the `traccc::memory_resource` given to the algorithm constructor. (Remember that `vecmem::sycl::copy` inherits from `vecmem::copy`.)

Reply:
I thought so too but I tested it on CUDA backend and it seemed to work, that's why I kept only `vecmem::sycl::copy`. But I also don't know why now it would work :stuck_out_tongue: so you're right, I will change make this distinction for `shared_mr`.